assume/common/utils.py

```diff
@@ -324,29 +324,36 @@ def get_test_demand_orders(power: np.array):
     return demand_order
 
 
-def separate_block_orders(orderbook):
+def separate_orders(orderbook):
+    # separate orders with several hours into single hour orders
+    delete_orders = []
     for order in orderbook:
-        if "bid_type" not in order.keys():
-            continue
-        elif order["bid_type"] == "BB" and isinstance(order["volume"], dict):
+        if True in [isinstance(order[key], dict) for key in order.keys()]:
             start_hour = order["start_time"]
             end_hour = order["end_time"]
-            duration = (end_hour - start_hour) / len(order["volume"])
+            duration = (end_hour - start_hour) / max(
+                len(order[key]) for key in order.keys() if isinstance(order[key], dict)
```

Comment:
But before I just checked if there are any dicts and now I need the amx length to determine the start and end times. With the check before it is ensured that there is at least one dict, so the else case is not needed

Reply:
so:
`(len(order[key]) for key in order.keys() if isinstance(order[key], dict))`
is equivalent to:
`max(len(value) for value in order.values() if isinstance(value, dict))`

So you need to have at least one entry in your order which is a dict - otherwise you would have `max([])`

For me it is unclear what that code does :D

Reply:
I would have expected something like 
```
order_len = len(order["volume"]) if isinstance(order["volume"], dict) else 1
duration = (end_hour - start_hour) / oder_len
```

Reply:
'max(len(value) for value in order.values() if isinstance(value, dict))'
This line is definetely nicer...
The reason why I didn't use len(order["volume"]) is to make it more general, but I guess we could also ensure that at least the volume has to be a dict, if its not a 'SB'